answers/extraction.py

```diff
@@ -0,0 +1,94 @@
+"""
+Handles the parsing and extraction of information from rST doc files.
+"""
+
+import os
+
+import docutils
+import docutils.nodes
+import docutils.parsers.rst
+
+# This is the variable in which all of the documentation will be parsed and
+# stored.
+# Documentation of section x-y-z is stored in DATA['x-y-z']
+DATA = {}
+
+
+class IgnoredDirective(docutils.parsers.rst.Directive):
+
+    """Stub for unknown directives."""
+
+    has_content = True
+
+    def run(self):
+        """Do nothing."""
+        return []
+
+docutils.parsers.rst.directives.register_directive('seealso', IgnoredDirective)
+docutils.parsers.rst.directives.register_directive('argparse', IgnoredDirective)
+
+class Extractor(docutils.nodes.SparseNodeVisitor):
+    """
+    Node visitor to extract information from nodes.
+    """
+
+    def __init__(self, document, name=''):
+        super().__init__(document)
+        self.name = name
+
+    def visit_section(self, node):
+        non_section_childs = list(filter(
+            lambda x: type(x) != docutils.nodes.section, node.children
+        ))
+        handle_non_section_nodes(node, non_section_childs, self.name)
+
+def parse_rst(path):
+    """
+    :param path: The path of the rst file.
+    :return: The document object
+    """
+    rst = open(path)
+    default_settings = docutils.frontend.OptionParser(
+        components=(docutils.parsers.rst.Parser, )
+        ).get_default_values()
+    document = docutils.utils.new_document(rst.name, default_settings)
+    parser =  docutils.parsers.rst.Parser()
+    parser.parse(rst.read(), document)
+    rst.close()
+    return document
+
+def handle_non_section_nodes(section_node, non_section_child_nodes, doc_name):
+    """
+    All the ndoes that are not section nodes are parsed here.
+    """
+    non_code_nodes = filter(
+        lambda x: type(x) not in [docutils.nodes.literal_block],
```

Comment:
isinstance would require the class as well. IMO this is better than `all([not isinstance(x, y) for y in []])`, no?